jormungandr/src/fragment/pool.rs

```diff
@@ -86,8 +86,10 @@ impl Pool {
     }
 
     pub async fn poll_purge(&mut self) -> Result<(), time::Error> {
-        let mut pool = self.pool.lock().await;
-        future::poll_fn(move |cx| pool.poll_purge(cx)).await?;
+        {
+            let mut pool = self.pool.lock().await;
+            future::poll_fn(move |cx| pool.poll_purge(cx)).await?;
+        }
```

Comment:
what about adding a `std::mem::drop(pool)` too to make it more obvious ?

Reply:
I prefer scoping since it demarcates the lifetime boundaries better and is less prone to refactoring errors.